Documentation/ExtensionArchitecture/Extbase/Reference/FileUpload.rst

```diff
@@ -0,0 +1,502 @@
+was only possible by either:
+
+*  Manually evaluate the :php:`$_FILES` data, process and validate the data,
+   use raw QueryBuilder write actions on :sql:`sys_file` and :sql:`sys_file_reference`
+   to persist the files quickly, or use at least some API methods:
+
+   ..  literalinclude:: _FileUpload/_ApiUpload.php
+       :caption: EXT:my_extension/Classes/Controller/BlogController.php, excerpt
+
+   Instead of raw access to :php:`$_FILES`, starting with TYPO3 v12 the recommendation
+   is to utilize the :ref:`UploadedFile objects instead of $_FILES <changelog:breaking-97214>`.
+   In that case, validators can be used for custom UploadedFile objects to specify restrictions
+   on file types, file sizes and image dimensions.
+
+*  Use (or better: adapt) a more complex implementation by using Extbase TypeConverters,
+   as provided by `Helmut Hummel's EXT:upload_example <https://github.com/helhum/upload_example>`__.
+   This extension is no longer maintained and will not work without larger adaptation for
+   TYPO3 v12 compatibility.
+
+..  _extbase_fileupload_writing-attributes:
+
+Automatic handling based on PHP attributes
+..........................................
+
+Starting with TYPO3 v13.3 it is finally possible to streamline this with commonly
+know Extbase logic, as implemented via
+:ref:`Feature: #103511 - Introduce Extbase file upload and deletion handling <changelog:feature-103511-1711894330>`.
+
+An example implementation of this can be found on
+`Torben Hansen's EXT:extbase-upload <https://github.com/derhansen/extbase_upload>`__
+repository.
+
+The general idea is to use PHP attributes within the Extbase Model, and for the upload
+use a custom ViewHelper.
+
+The domain model:
+
+..  literalinclude:: _FileUpload/_BlogEnhanced.php
+    :caption: EXT:my_extension/Classes/Domain/Model/Blog.php, using FileUpload attributes
+
+and the corresponding Fluid template utilizing the ViewHelper:
+
+..  literalinclude:: _FileUpload/_Upload.html
+    :caption: EXT:my_extension/Resources/Private/Templates/Blog/New.html
+
+You can also allow to remove already uploaded files (for the user):
+
+..  literalinclude:: _FileUpload/_UploadDelete.html
+    :caption: EXT:my_extension/Resources/Private/Templates/Blog/New.html
+
+The controller action part with persisting the data needs no further custom code,
+Extbase can automatically do all the domain model handling on its own. The TCA can
+also stay the same as configured for simply read-access to a domain model.
+
+The actual file upload processing is performed just before Extbase persists a
+domain model Entity, so only if the full validation has taken place and no errors
+occurred. This means, if any error occurs, a user will have to re-upload a file.
+
+The implementation is done like this to prevent stale temporary files that would
+need cleanup or could raise issues with Denial of Service (by filling up disk-space
+with these temporarily uploaded files).
+
```

Comment:
Please add:

```
.. important::
   
   When working with file uploads in domain models, it is required to persist the 
   model within the same request (e.g. call :php:`$myRespository->add($myModel)`)
   method in the target action in order to prevent stale temporary files.
```

Reply:
done